src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -759,19 +759,19 @@ private void configureTableCommands() {
 			boolean canEditResults = entityBundle.getPermissions().getCanCertifiedUserEdit() && !isDataset;
 			actionMenu.setActionVisible(Action.UPLOAD_TABLE_DATA, canEditResults);
 			actionMenu.setActionText(Action.UPLOAD_TABLE_DATA, "Upload Data to " + entityTypeDisplay);
-			actionMenu.setActionVisible(Action.EDIT_TABLE_DATA, canEditResults);
 			actionMenu.setActionVisible(Action.DOWNLOAD_TABLE_QUERY_RESULTS, true);
 			actionMenu.setActionVisible(Action.SHOW_TABLE_SCHEMA, true);
 			actionMenu.setActionVisible(Action.SHOW_VIEW_SCOPE, !(entityBundle.getEntity() instanceof TableEntity) && !isDataset);
 			actionMenu.setActionVisible(Action.EDIT_DATASET_ITEMS, isDataset);
 		} else {
 			actionMenu.setActionVisible(Action.UPLOAD_TABLE_DATA, false);
-			actionMenu.setActionVisible(Action.EDIT_TABLE_DATA, false);
 			actionMenu.setActionVisible(Action.DOWNLOAD_TABLE_QUERY_RESULTS, false);
 			actionMenu.setActionVisible(Action.SHOW_TABLE_SCHEMA, false);
 			actionMenu.setActionVisible(Action.SHOW_VIEW_SCOPE, false);
 			actionMenu.setActionVisible(Action.EDIT_DATASET_ITEMS, false);
 		}
+		// The EDIT_TABLE_DATA action will be shown by other widgets when data is editable.
+		actionMenu.setActionVisible(Action.EDIT_TABLE_DATA, false);
```

Comment:
Whether data is editable depends on the query state. If a View query has no results (or we don't even query it because it has no scope/columns), then we need to alter the state of this button, which we cannot do here.